src/VirtualClient/VirtualClient.Dependencies/CUDAAndNvidiaGPUDriverInstallation.cs

```diff
@@ -184,7 +184,7 @@ await this.stateManager.SaveStateAsync(nameof(CudaAndNvidiaGPUDriverInstallation
                     await this.CudaAndNvidiaGPUDriverInstallationOnWindowsAsync(telemetryContext, cancellationToken)
                                .ConfigureAwait(false);
 
-                    await this.stateManager.SaveStateAsync(nameof(this.CudaAndNvidiaGPUDriverInstallationOnWindowsAsync), new State(), cancellationToken)
+                    await this.stateManager.SaveStateAsync(nameof(CudaAndNvidiaGPUDriverInstallation), new State(), cancellationToken)
```

Comment:
Did this break on you? This should work if you are only working on Windows and not switching OS

Reply:
on windows, if I run multiple iterations of installGPUDriver, the first iteration will install the driver successfully, but the following iterations will not skip the installation as expected. I saw that on [line 147](https://github.com/microsoft/VirtualClient/blob/b91e115e177d733b8911a4fc0175b420e08be7cc/src/VirtualClient/VirtualClient.Dependencies/CUDAAndNvidiaGPUDriverInstallation.cs#L147), the executor is getting the state using the name CudaAndNvidiaGPUDriverInstallation, not CudaAndNvidiaGPUDriverInstallationOnWindowsAsync. After changing the variable name, VC skips execution after the first run as expected.